libcxx/test/std/containers/associative/map/map.access/at.pass.cpp

```diff
@@ -93,6 +102,23 @@ int main(int, char**) {
         V(7, 7.5),
         V(8, 8.5),
     };
+
+    // std::__tree_node<std::__value_type<int,double>, min_pointer<void>> d;
+    // std::__tree_node_base<min_pointer<void>> b  = d;
+    using Base = std::__tree_node_base<min_pointer<void>>;
+
+    using Derived = std::__tree_node<std::__value_type<int, double>, min_pointer<void>>;
+    static_assert(std::is_base_of_v<Base, Derived>);
+
+    // using BaseP = min_pointer<Base>;
+    // using DerivedP = min_pointer<Derived>;
+    // static_assert(std::is_base_of_v<BaseP, DerivedP>);
+    // DerivedP dp(nullptr);
+    // (void)dp;
+
+    // BaseP bp =static_cast<BaseP>(dp);
+    // (void)bp;
```

Comment:
You can't use implementation details in standard tests (and here I also don't see the point).

Reply:
Will clean it up, this was before I realized the "fix" was using static fancy pointer cast over static_cast